smithy-diff/src/main/java/software/amazon/smithy/diff/evaluators/ChangedEnumTrait.java

```diff
@@ -93,10 +96,17 @@ private List<ValidationEvent> validateEnum(ChangedShape<Shape> change, Pair<Enum
         for (EnumDefinition definition : newTrait.getValues()) {
             if (!oldTrait.getEnumDefinitionValues().contains(definition.getValue())) {
                 if (newPosition <= oldEndPosition) {
-                    events.add(error(change.getNewShape(), String.format(
-                            "Enum value `%s` was inserted before the end of the list of existing values. This can "
-                                    + "cause compatibility issues when ordinal values are used for iteration, "
-                                    + "serialization, etc.", definition.getValue())));
+                    events.add(
+                            ValidationEvent.builder()
+                                    .severity(Severity.ERROR)
+                                    .message(String.format(
+                                    "Enum value `%s` was inserted before the end of the list of existing values. This "
+                                            + "can cause compatibility issues when ordinal values are used for "
+                                            + "iteration, serialization, etc.", definition.getValue()))
+                                    .shape(change.getNewShape())
+                                    .id(getEventId() + "." + INSERTED)
```

Comment:
Nit, let's move the "." part to the INSERTED constant, and use here just `getEventId() + INSERTED`. Also I think that "OrderChanged" would be better.

Can we also add specificity to the other cases that error here, like `Removed` and `Changed`?

Reply:
Ack, addressing all suggestions in next commit